prospero-common/src/main/java/org/wildfly/prospero/actions/ApplyStageBackup.java

```diff
@@ -65,55 +67,106 @@ class ApplyStageBackup implements AutoCloseable {
     }
 
     /**
-     * add all the files in the server to cache
+     * add all the server-managed files in the server to cache
      *
      * @throws IOException - if unable to backup the files
      */
     public void recordAll() throws IOException {
-        Files.walkFileTree(serverRoot, new SimpleFileVisitor<>() {
+        ProsperoLogger.ROOT_LOGGER.debug("Starting building the update backup.");
+        final Path hashesRoot = serverRoot.resolve(Constants.PROVISIONED_STATE_DIR).resolve(Constants.HASHES);
 
-            @Override
-            public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
-                // if the file in the installation is not readable AND it does not exist in the candidate
-                // we assume it is user controlled file and we ignore it
-                return ignoreUserManagedFiles(file, exc);
-            }
+        // if the server doesn't conatain a galleon provisioning record, skip building the backup
+        if (!Files.exists(hashesRoot)) {
+            ProsperoLogger.ROOT_LOGGER.warn("Unable to perform the backup: No Galleon Hashes record found.");
```

Comment:
Are we at risk that this version of prospero will by used with an instance provisioned by an old prospero instance? In which case it would become unusable?

Reply:
Do you mean if the galleon `.provisioned/hashes` folder is not present? That folder should be present in any instance provisioned by Prospero. Or is there some other change that might be problematic?

If the hashes folder doesn't exist, the update should still proceed, just without the backup. Which actually I should add to the testsuite :)

Reply:
Yeah, I'm missing that context, I thought the hashes file is a new thing. This looks OK then :).